src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -739,7 +739,7 @@ internal static class CoreTypes
                     { typeof(Guid),                                        new[] { "guid" } },
                     { typeof(Hashtable),                                   new[] { "hashtable" } },
                     { typeof(int),                                         new[] { "int", "int32" } },
-                    { typeof(Int16),                                       new[] { "int16" } },
+                    { typeof(Int16),                                       new[] { "short", "int16" } },
                     { typeof(long),                                        new[] { "long", "int64" } },
```

Comment:
Script portability might suffer a bit by adding these type accelerators. I can see C# developers adopting them, and  older versions of PowerShell obviously don't have them. I think there should be some consensus that adding these is a worth it.

Reply:
Can you elaborate on why portability might suffer? Are you referring more to backportability?

Reply:
Gotcha. I can see that being somewhat of a concern on one hand, but I also feel that if we avoid making QOL changes just because someone may still be on an older version of PS doesn't... really hold that much weight. 

We're talking about PS Core here, after all, and there are already several things similarly basically just quality-of-life changes that have been included. Example that I can think of right this minute would be the character support for range operators. `'a'..'z'` doesn't work in WinPS on any version at all.

Certainly, it may hurt backportability slightly, but this does not remove the existing type accelerators -- if one wishes to use those instead (i.e., `[uint32]` instead of `[uint]`) for compatibility reasons, there is nothing stopping you from doing so for compatibility concerns.

As with many other aspects of PS, if you need to write scripts for older variants of PS, you can test them against those old versions first. Few scripts written in PS 5.1 that I come across are instantly compatible with v3, often requiring some small changes, and forget about v2 entirely. This would simply become another of those things (like the char support for range operator) that you have to spend a little extra typing on if you're concerned about backportability.

Reply:
I think adding this functionality is a welcome move, it does add quite a bit of flexibility.